components/TARGET_PSA/services/crypto/COMPONENT_SPE/psa_crypto_partition.c

```diff
@@ -134,24 +140,57 @@ static void psa_mac_operation(void)
                 }
 
                 case PSA_MAC_UPDATE: {
-                    uint8_t *input_ptr = mbedtls_calloc(1, msg.in_size[1]);
-                    if (input_ptr == NULL) {
+
+                    uint8_t * input_buffer = NULL;
+                    size_t data_remaining = msg.in_size[1];
+                    size_t allocation_size = data_remaining;
+                    size_t size_to_read = 0;
+                    psa_status_t failure_status = PSA_SUCCESS;
+                    
+                    if (allocation_size > MAX_DATA_CHUNK_SIZE)
+                    {
+                        allocation_size = MAX_DATA_CHUNK_SIZE;
+                    }
+
+                    input_buffer = mbedtls_calloc(1, allocation_size);
+                    if (input_buffer == NULL) {
                         status = PSA_ERROR_INSUFFICIENT_MEMORY;
                         break;
                     }
 
-                    bytes_read = psa_read(msg.handle, 1, input_ptr,
-                                          msg.in_size[1]);
+                    while (data_remaining > 0)
+                    {
+                        size_to_read = data_remaining;
+                        if (size_to_read > MAX_DATA_CHUNK_SIZE)
+                        {
+                            size_to_read = MAX_DATA_CHUNK_SIZE;
+                        }
+                        bytes_read = psa_read(msg.handle, 1, input_buffer,
+                                              size_to_read);
 
-                    if (bytes_read != msg.in_size[1]) {
-                        SPM_PANIC("SPM read length mismatch");
-                    }
+                        if (bytes_read != size_to_read) {
+                            SPM_PANIC("SPM read length mismatch");
+                        }
 
-                    status = psa_mac_update(msg.rhandle,
-                                            input_ptr,
-                                            msg.in_size[1]);
+                        status = psa_mac_update(msg.rhandle,
+                                                input_buffer,
+                                                bytes_read);
+                        
+                        // return first error encountered in case of error.
+                        if ((status != PSA_SUCCESS) && 
```

Comment:
if you have an error, just break from the loop and return the initial error status.
no need to have failure_status

Reply:
@alzix , @avolinski : ok, since all of you don't think this behavior is not required I'll remove it pending security review.